src/lib/proposer/proposer.ml

```diff
@@ -79,472 +47,452 @@ end = struct
     interruptible
 end
 
-module Make (Inputs : Inputs_intf) :
```

Comment:
Would we actually want to completely defunctor proposer? Wouldn't we want to create some unit tests for it sometime in the future?

Reply:
If we have unit tests in the future, we can always add a small functor back in for anything we need to mock. It's easier to refunctor than it is to defunctor. Most of our functors that exist for tests stub out more inputs then they actually require custom mocks for too, so it's better to have only the required inputs you need for your tests.